core/templates/dev/head/domain/exploration/AnswerGroupObjectFactory.js

```diff
@@ -0,0 +1,30 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+oppia.factory('AnswerGroupObjectFactory', [
+  function() {
+    var AnswerGroup = function(ruleSpecs, outcome) {
+      this.rule_specs = ruleSpecs;
```

Comment:
I'm having a bit of trouble with this. The tests keep failing, I suspect due to the tests creating opaque JS dicts as AnswerGroups. Would the correct approach to resolve this be to inject the factory into the tests and use it in the tests? Sorry haven't really used jasmine before so am not sure of the right approach